decidim-censuses/app/controllers/decidim/censuses/admin/census_uploads_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Decidim
+  module Censuses
+    module Admin
+      class CensusUploadsController < Decidim::Admin::ApplicationController
+
+        before_action :show_instructions, unless: :module_active?
+
+        def show
+          authorize! :show, Census
+          @status = Status.new
+        end
+
+        def create
+          authorize! :create, Census
+          if params[:file]
+            data = CsvData.new(params[:file].path)
+            Census.insert_all data.values
+            RemoveDuplicatesJob.perform_later
+            flash[:notice] = t('.success', count: data.values.count,
+                                           errors: data.errors.count)
+          end
+          redirect_to census_uploads_path
+        end
+
+        def delete_all
+          authorize! :destroy, Census
+          Census.delete_all
+          flash[:notice] = t('.success')
+          redirect_to census_uploads_path
```

Comment:
you can use: `redirect_to census_uploads_path, notice: t('.success')` which is shorter.